File: market/test/java/ru/yandex/market/logistics/management/domain/entity/type/EnumMappingTest.java
package ru.yandex.market.logistics.management.domain.entity.type;

import java.util.Set;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import ru.yandex.market.logistics.management.domain.dto.front.logistic.segment.LogisticSegmentFrontType;
import ru.yandex.market.logistics.management.domain.dto.front.partner.PartnerTypeFrontEnum;
import ru.yandex.market.logistics.management.entity.type.LogisticSegmentType;

import static org.junit.jupiter.api.Assertions.assertEquals;

class EnumMappingTest {

    @ParameterizedTest
    @MethodSource("getMappedEnums")
    void testEnumsMapping(Class<? extends Enum> enum1, Class<? extends Enum> enum2) {
        assertEquals(
            toStringSet(enum1),
            toStringSet(enum2)
        );
    }

    private static Set<String> toStringSet(Class<? extends Enum> enum1) {
        return Stream.of(enum1.getEnumConstants())
            .map(Enum::name)
            .collect(Collectors.toSet());
    }

    private static Stream<Arguments> getMappedEnums() {
        return Stream.of(
            Arguments.of(
                PossibleOrderChangeType.class,
                ru.yandex.market.logistics.management.entity.type.PossibleOrderChangeType.class
            ),
            Arguments.of(
                PossibleOrderChangeMethod.class,
                ru.yandex.market.logistics.management.entity.type.PossibleOrderChangeMethod.class
            ),
            Arguments.of(
                TrackCodeSource.class,
                ru.yandex.market.logistics.management.entity.type.TrackCodeSource.class
            ),
            Arguments.of(
                LogisticSegmentType.class,
                LogisticSegmentFrontType.class
            ),
            Arguments.of(
                PartnerType.class,
                PartnerTypeFrontEnum.class
            )
        );
    }
}

File: market/test/kotlin/ru/yandex/market/logistics/les/EnumMappingTest.kt
package ru.yandex.market.logistics.les

import io.kotest.assertions.assertSoftly
import io.kotest.matchers.shouldBe
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import ru.yandex.market.logistics.les.admin.model.enums.AdminEntityType
import ru.yandex.market.logistics.les.base.EntityType

class EnumMappingTest {

    @ParameterizedTest
    @MethodSource("getMappedEnums")
    fun testEnumsMapping(enum1: Class<out Enum<*>>, enum2: Class<out Enum<*>>) {
        assertSoftly {
            toStringSet(enum1) shouldBe toStringSet(enum2)
        }
    }

    companion object {
        @JvmStatic
        fun getMappedEnums() = listOf(
            Arguments.of(
                EntityType::class.java,
                AdminEntityType::class.java
            )
        )
    }

    private fun toStringSet(enum: Class<out Enum<*>>) = enum.enumConstants.map { it.name }.toSet()
}
